### DISCUSSION TITLE
Checkbox all is selecting all rows instead the rows on the same page

### DISCUSSION THREAD

**Discussion by paramwork** [rappasoft/laravel-livewire-tables#1022] [Q&A] [answered]
Hi,

I recently upgraded from v1. On v1, when I click the select checkbox it only selects the rows from the same page.
On, v2 when I click the select checkbox it selects all rows from all pages causing it to use too much memory.

Is there a way to limit the selection of rows in version 2 to only those on the current page, similar to how it worked in version 1 when clicking on the select checkbox, as it is causing an issue with memory usage by selecting all rows across all pages?

**Comment by lrljoe** [comment]
Currently?  No.  

I'll create the relevant config for this over the weekend and issue a pull request. But it'll be a little wait before it is part of the core repo.

**Comment by lrljoe** [comment]
As promised I have written the code for this, and opened a pull request, @rappasoft will take a look when he gets time I'm sure!

It is a very straight forward change, so you could implement it yourself in the interim looking at the code in my commit, or you can clone my dev-master in the interim if you'd prefer (see details on my fork), as that has a number of recent pull requests implemented, just make sure to switch back once @rappasoft has time to update the core!

[View Pull Request](https://github.com/rappasoft/laravel-livewire-tables/pull/1023)

**Reply by rappasoft** [reply]
I appreciate all your work on this repo believe me. I'm doing my best to look at all of it this week!

**Reply by lrljoe** [reply]
No problem at all!  It's a great package that I get a lot of value out of, so happy to contribute.  if you think I'm missing anything in my pull requests or want me to adjust the coding style at all, just let me know.

**Comment by lrljoe** [comment]
@paramwork - I think you can expect this pull request to be fixed wtihin the next week or two within core.

**Comment by lrljoe** [comment]
Quick update, I managed to stuff up my various branches, so please wait caller, I'm hoping to have a chance to sort this weekend.

**Comment by lrljoe** [comment]
See PR #1112 for a fix for this, it adds in a Select All On Page button.

**Comment by lrljoe** [comment, marked as answer]
Due to the sheer number of changes I've put in elsewhere, this will be in the release after next unless something pushes the date back!

**Reply by paramwork** [reply]
Thanks for doing it. I have moved to Filament Table package for better performance.

**Reply by lrljoe** [reply]
First thanks for the update and marking as resolved!!

Out of interest, where are you seeing performance gains?  Just in case there's some obvious areas that need improving on this package.
